parcels/particlesets/particlesetsoa.py

```diff
@@ -0,0 +1,675 @@
+        #                 self.particle_data[v.name][i] = v.initial[
+        #                     time[i], depth[i], lat[i], lon[i]
+        #                 ]
+        #                 logger.warning_once("Particle initialisation from field can be very slow as it is computed in scipy mode.")
+        #         elif isinstance(v.initial, attrgetter):
+        #             self.particle_data[v.name][:] = v.initial(self)
+        #         else:
+        #             self.particle_data[v.name][:] = v.initial
+
+        #         initialised.add(v.name)
+        # else:
+        #     raise ValueError("Latitude and longitude required for generating ParticleSet")
+
+    def _set_particle_vector(self, name, value, indices=None):
+        """Set attributes of all particles to new values.
+
+        :param name: Name of the attribute (str).
+        :param value: New value to set the attribute of the particles to.
+        :param indices: (Optional) only set the particles with these indices.
+                        Its length should be equal to the length of 'values'.
+                        If None, all particles are set.
+        """
+
+        if indices is None:
+            self.collection._data[name][:] = value
+        else:
+            self.collection._data[name][indices] = value
+
+    def _get_particle_vector(self, name, indices=None):
+        """Set attributes of all particles to new values.
+
+        :param name: Name of the attribute (str).
+        :param indices: (Optional) only set the particles with these indices.
+                        Its length should be equal to the length of 'values'.
+                        If None, all particles are set.
+        :return: The values of the particle attributes.
+        """
+        if indices is None:
+            return self.collection._data[name]
+        else:
+            return self.collection.data[name][indices]
+
+    def _impute_release_times(self, default):
+#         np.nan_to_num(self._collection._data['time'], nan=default)
+        self._collection._data['time'][
+            np.isnan(self._collection._data['time'])] = default
+        return (np.min(self._collection._data['time']),
+                np.max(self._collection._data['time']))
+
+    @property
+    def error_particles(self):
+        error_indices = np.where(np.isin(
+            self.collection._data['state'],
+            [StateCode.Success, StateCode.Evaluate], invert=True))[0]
+        return ParticleCollectionIteratorSOA(self._collection,
+                                             subset=error_indices)
+
+    def __getattr__(self, name):
+        # Comment CK: this either a member function of the accessor or the collection - not the PSet itself
+        if name in self._collection._data:
+            return getattr(self._collection, name)
+        elif name in self.__dict__:
+            return self.__dict__[name]
+        else:
+            return False
```

Comment:
This is a point we have to discuss a bit more. We talked about it previously @CKehl. Anyway, it is not just the tests that use notation like `pset.lon`, there are user-use-cases as well (e.g. plotting trajectories). So it is probably worth keeping this notation in place (at least for now), even though it does not have an obvious efficient implementation in every imaginable datastructure.
I suggest we discuss this this afternoon, but I wanted to have this comment in place to refer to for the discussion :)
If we keep it this function should probably move up to the BaseParticleSet and the implementation of __getattr__ should be required by the Collection interface.

Reply:
I think we resolved it in the DevTalk today.